diff --git a/android/build.gradle b/android/build.gradle
index 8fce879dcf7a0d92209550aebde7be63b0aba530..0560ec2d2a7745dbcf5d5c914f92b6a14e926f60 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,30 +1,16 @@
 apply plugin: 'com.android.library'
 
-repositories {
-    mavenLocal()
-    jcenter()
-    maven {
-        // For developing the library outside the context of the example app, expect `react-native`
-        // to be installed at `./node_modules`.
-        url "$projectDir/../node_modules/react-native/android"
-    }
-    maven {
-        // For developing the example app.
-        url "$projectDir/../../react-native/android"
-    }
-}
-
-def DEFAULT_COMPILE_SDK_VERSION = 28
-def DEFAULT_BUILD_TOOLS_VERSION = "28.0.3"
-def DEFAULT_TARGET_SDK_VERSION = 28
+def DEFAULT_COMPILE_SDK_VERSION = 34
+def DEFAULT_BUILD_TOOLS_VERSION = "34.0.0"
+def DEFAULT_TARGET_SDK_VERSION = 34
 def DEFAULT_SUPPORT_LIB_VERSION = "28.0.0"
 
 android {
-    compileSdkVersion rootProject.hasProperty('compileSdkVersion') ? rootProject.compileSdkVersion : DEFAULT_COMPILE_SDK_VERSION
+    compileSdk DEFAULT_COMPILE_SDK_VERSION
     buildToolsVersion rootProject.hasProperty('buildToolsVersion') ? rootProject.buildToolsVersion : DEFAULT_BUILD_TOOLS_VERSION
 
     defaultConfig {
-        minSdkVersion 15
+        minSdkVersion 24
         targetSdkVersion rootProject.hasProperty('targetSdkVersion') ? rootProject.targetSdkVersion : DEFAULT_TARGET_SDK_VERSION
         versionCode 1
         versionName "1.0"
@@ -37,26 +23,6 @@ android {
     }
 }
 
-buildscript {
-    if (project == rootProject) {
-        repositories {
-            jcenter()
-        }
-        dependencies {
-            classpath 'com.android.tools.build:gradle:3.3.2'
-
-            // NOTE: Do not place your application dependencies here; they belong
-            // in the individual module build.gradle files
-        }
-    }
-}
-
-allprojects {
-    repositories {
-        jcenter()
-    }
-}
-
 def supportVersion = rootProject.hasProperty('supportLibVersion') ? rootProject.supportLibVersion : DEFAULT_SUPPORT_LIB_VERSION
 
 dependencies {
